frappe/utils/doctor.py

```diff
@@ -33,118 +13,47 @@ def purge_pending_tasks(event=None, site=None):
-
-	count = 0
+	purged_task_count = 0
+	for queue in get_queue_list():
+		q = get_queue(queue)
+		purged_task_count += q.count
+		q.empty()
 
-	for queue in get_queues():
-		for taskstr in r.lrange(queue, 0, -1):
-			taskbody = get_task_body(taskstr)
-			kwargs = taskbody.get('kwargs')
-			if kwargs:
-				if site and kwargs.get('site') != site:
-					continue
-
-				if event:
-					if kwargs.get('handler') and kwargs.get('handler') in event_tasks:
-						r.lrem(queue, taskstr)
-						count += 1
-				else:
-					r.lrem(queue, taskstr)
-					count += 1
-	return count
+	return purged_task_count
 
 def get_pending_task_count():
 	"Get count of pending tasks"
-	r = get_redis_conn()
 	pending = 0
-	for queue in get_queues():
-		pending += r.llen(queue)
+	for queue in get_queue_list():
+		q = get_queue(queue)
+		for job in q.jobs:
+			pending += 1
 	return pending
 
-def get_timedout_locks():
-	"Get list of stale locks from all sites"
-	old_locks=[]
-	for site in frappe.utils.get_sites():
-		locksdir = os.path.join(site, 'locks')
-		for lock in os.listdir(locksdir):
-			lock_path = os.path.join(locksdir, lock)
-			try:
-				check_lock(lock_path)
-			except LockTimeoutError:
-				old_locks.append(lock_path)
-	return old_locks
 
 def check_if_workers_online():
-	app = get_celery()
-	if app.control.inspect().ping():
-		return True
-	return False
+	with Connection():
+		if Worker.all():
+			return True
+		else: 
+			return False
 
 def dump_queue_status(site=None):
 	"""
 	Dumps pending events and tasks per queue
 	"""
-	ret = []
-	r = get_redis_conn()
-	for queue in get_queues(site=site):
-		if r.llen(queue):
-			queue_details = {
-				'queue': queue,
-				'len': r.llen(queue),
-			}
-			queue_details.update(get_task_count_for_queue(queue))
-			ret.append(queue_details)
-
-	ret = sorted(ret, key=itemgetter('len'), reverse=True)
-	ret.insert(0, {
-		'total': get_pending_task_count()
-	})
-	return ret
+	return get_jobs(site=site)
 
 def get_task_count_for_queue(queue):
 	"""
 	For a given queue, returns the count of every pending task and aggregate of
 	events pending
 	"""
-	r = get_redis_conn()
-	tasks = [get_task_body(taskstr) for taskstr in r.lrange(queue, 0, -1)]
-	task_names = [task['task'] for task in tasks]
-	task_counts = Counter(task_names)
-	event_counts = Counter(task['kwargs'].get('event') for task in tasks)
-	return {
-		'task_counts': task_counts,
-		'event_counts': event_counts
-	}
-
+	return get_jobs(queue=queue)
+	
 def get_running_tasks():
-	ret = {}
-	app = get_celery()
-	inspect = app.control.inspect()
-	active = inspect.active()
-	if not active:
-		return []
-	for worker in active:
-		ret[worker] = []
-		for task in active[worker]:
-			ret[worker].append({
-				'id': task['id'],
-				'name': task['name'],
-				'routing_key': task['delivery_info']['routing_key'],
-				'args': task['args'],
-				'kwargs': task['kwargs']
-			})
-	return ret
+	url, port = get_redis_conn()
+	return get_current_job(connection=Redis())
```

Comment:
AFAIK this won't work. `get_current_job` works if the current process is a worker process